src/makeRobot.js

```diff
@@ -37,8 +37,67 @@
  *
  * @return {object}
  */
+
 function makeRobot(name, wheels, version) {
-  // write code here
+  return {
+    name: name,
+    wheels: wheels,
+    version: version,
+    coords: {
+      x: 0,
+      y: 0,
+    },
+
+    get info() {
+      return `name: ${this.name}, `
+        + `chip version: ${this.version}, wheels: ${this.wheels}`;
+    },
+    get location() {
+      return `${this.name}: x=${this.coords.x}, y=${this.coords.y}`;
+    },
+
+    goForward(distance = 1) {
+      if (distance < 0) {
+        return this;
+      }
+      this.coords.y += distance;
+
+      return this;
+    },
+
+    goBack(distance = 1) {
+      if (distance > 0) {
+        this.coords.y -= distance;
+      }
+
+      return this;
+    },
+
+    goRight(distance = 1) {
+      if (distance < 0) {
+        return this;
+      }
+      this.coords.x += distance;
+
+      return this;
+    },
+
+    goLeft(distance = 1) {
+      if (distance < 0) {
+        return this;
+      }
+      this.coords.x -= distance;
+
+      return this;
+    },
+
+    evacuate() {
+      this.coords.x = 1400;
+      this.coords.y = 500;
+
+      return this;
+    },
```

Comment:
`return this` is not necessary here
```suggestion
    evacuate() {
      this.coords.x = 1400;
      this.coords.y = 500;
    },
```